components/studio/bin/hab-studio.sh

```diff
@@ -1076,6 +1076,35 @@ chown_artifacts() {
   fi
 }
 
+# **Internal**  Run a command which may fail without aborting whole script
+try() {
+  local status
+  if [[ $- == *e* ]]; then
```

Comment:
You'd need to use `if [ $- = *e* ]; then` here to make sure it would work outside of Bash--I *think* the wildcard match would still work though.

Reply:
I don't think that works with [single brackets](http://www.tldp.org/LDP/abs/html/comparison-ops.html#SCOMPARISON1), unfortunately. But, I think your later comment obviates the need to mess with `-e`.